src/test/java/edu/ut/ece/social/hw1/SampleAnswerTest.java

```diff
@@ -0,0 +1,112 @@
+package edu.ut.ece.social.hw1;
+
+import com.google.common.collect.ImmutableList;
+import edu.ut.ece.social.hw1.dgs.DGS;
+import edu.ut.ece.social.hw1.km.KM;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.io.FileNotFoundException;
+import java.util.Collection;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(Parameterized.class)
+public class SampleAnswerTest {
+
+
+    @Parameters
```

Comment:
How does the class know to run all the items in the collection?

Reply:
Because this test now runs with the `Parameterized` test runner instead of the standard `JUnit4` test runner. It looks for the collection of objects annotated as `@Parameters` and injects those into the constructor, instantiating the test once for each parameter.